### PR TITLE
Creating Entities Classes

### PR DESCRIPTION
**Pull request by ziadnasser19**

### CODE REVIEW

File: src/Backend/Entities/Department.java
Comment (BlindFl0W) [inline src/Backend/Entities/Department.java:1]: Looks Good.

File: src/Backend/Entities/Employee.java
Comment (BlindFl0W) [inline src/Backend/Entities/Employee.java:1]: Looks Good.

File: src/Backend/Entities/Supervisor.java [hunk 1]
```diff
@@ -1,7 +1,42 @@
...
+        IAttendanceCommand editAttendanceCommand = new EditAttendance(, ID, updatedRecord);
+        editAttendanceCommand.execute();
+    }
+
+    /**
+     * Method to view attendance history of employees.
+     * @param ID The employee ID whose attendance history is to be viewed.
+     */
+    public void superViewAttendanceHistory(int ID) {
+        IAttendanceCommand superViewAttendanceHistoryCommand = new SuperViewAttendanceHistory(getAttendanceManager(), ID);
```
Comment (BlindFl0W) [inline src/Backend/Entities/Supervisor.java:40]: And here

File: src/Backend/Entities/Supervisor.java [hunk 2]
```diff
@@ -1,7 +1,42 @@
...
+    public Supervisor(String name, int id, String position, Department department) {
+        super(name, id, position, department);
+    }
+
+    /**
+     * Method to edit attendance records for employees.
+     * @param ID The employee ID whose attendance record needs to be edited.
+     * @param updatedRecord The updated attendance record.
+     */
+    public void editAttendance(int ID, AttendanceRecord updatedRecord) {
```
Comment (BlindFl0W) [inline src/Backend/Entities/Supervisor.java:30]: Here

File: src/Backend/Entities/Supervisor.java
Comment (BlindFl0W) [inline src/Backend/Entities/Supervisor.java:1]: This need an edit.
Still waiting for the changes to the attendance command interfaces to make the edits.

**Review by BlindFl0W — CHANGES_REQUESTED** [review]
Still waiting for the changes to the attendance command interfaces to make the edits.

File: src/Backend/Entities/Supervisor.java
Comment (mistaluai) [reply src/Backend/Entities/Supervisor.java:1]: can you provide more details?

**Review by mistaluai — CHANGES_REQUESTED** [review]
provide an updated version according to amr's review.

**Pull request closed without merging**